Subject: [PATCH] iproute2: de-bash scripts

de-bash these two scripts to make iproute2 not depend on bash.

Upstream-Status: Pending

Signed-off-by: Chen Qi <Qi.Chen@windriver.com>

---
 ip/ifcfg |   15 ++++++++-------
 ip/rtpr  |    2 +-
 2 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/ip/ifcfg b/ip/ifcfg
index 083d9df..60bcf1f 100644
--- a/ip/ifcfg
+++ b/ip/ifcfg
@@ -1,12 +1,13 @@
-#! /bin/bash
+#! /bin/sh
 
 CheckForwarding () {
-  local sbase fwd
+  local sbase fwd forwarding
   sbase=/proc/sys/net/ipv4/conf
   fwd=0
   if [ -d $sbase ]; then
     for dir in $sbase/*/forwarding; do
-      fwd=$[$fwd + `cat $dir`]
+      forwarding=`cat $dir`
+      fwd=$(($fwd+$forwarding))
     done
   else
     fwd=2
@@ -127,12 +128,12 @@ fi
 arping -q -A -c 1 -I $dev $ipaddr
 noarp=$?
 ( sleep 2 ;
-  arping -q -U -c 1 -I $dev $ipaddr ) >& /dev/null </dev/null &
+  arping -q -U -c 1 -I $dev $ipaddr ) > /dev/null 2>&1 </dev/null &
 
-ip route add unreachable 224.0.0.0/24 >& /dev/null
-ip route add unreachable 255.255.255.255 >& /dev/null
+ip route add unreachable 224.0.0.0/24 > /dev/null 2>&1
+ip route add unreachable 255.255.255.255 > /dev/null 2>&1
 if [ `ip link ls $dev | grep -c MULTICAST` -ge 1 ]; then
-  ip route add 224.0.0.0/4 dev $dev scope global >& /dev/null
+  ip route add 224.0.0.0/4 dev $dev scope global > /dev/null 2>&1
 fi
 
 if [ $fwd -eq 0 ]; then
diff --git a/ip/rtpr b/ip/rtpr
index c3629fd..674198d 100644
--- a/ip/rtpr
+++ b/ip/rtpr
@@ -1,4 +1,4 @@
-#! /bin/bash
+#! /bin/sh
 
 exec tr "[\\\\]" "[
 ]"
-- 
1.7.9.5

